app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/AudioMediaRecorder.kt

```diff
@@ -268,41 +270,61 @@ class AudioMediaRecorder @Inject constructor(
                 }
 
                 val outputBufferIndex = codec.dequeueOutputBuffer(bufferInfo, TIMEOUT_US)
-                if (outputBufferIndex >= 0) {
-                    val outputBuffer = codec.getOutputBuffer(outputBufferIndex)
 
-                    if (bufferInfo.flags and MediaCodec.BUFFER_FLAG_CODEC_CONFIG != 0) {
-                        codec.releaseOutputBuffer(outputBufferIndex, false)
-                        continue
+                when {
+                    outputBufferIndex == MediaCodec.INFO_OUTPUT_FORMAT_CHANGED -> {
+                        val newFormat = codec.outputFormat
+                        trackIndex = muxer.addTrack(newFormat)
+                        muxer.start()
+                        retryCount = 0
                     }
+                    outputBufferIndex >= 0 -> {
+                        val outputBuffer = codec.getOutputBuffer(outputBufferIndex)
 
-                    if (bufferInfo.size != 0) {
-                        outputBuffer?.position(bufferInfo.offset)
-                        outputBuffer?.limit(bufferInfo.offset + bufferInfo.size)
-
-                        if (trackIndex == -1) {
-                            val newFormat = codec.outputFormat
-                            trackIndex = muxer.addTrack(newFormat)
-                            muxer.start()
+                        if (bufferInfo.flags and MediaCodec.BUFFER_FLAG_CODEC_CONFIG != 0) {
+                            codec.releaseOutputBuffer(outputBufferIndex, false)
+                            continue
                         }
 
-                        muxer.writeSampleData(trackIndex, outputBuffer!!, bufferInfo)
-                    }
+                        if (bufferInfo.size != 0 && outputBuffer != null) {
+                            outputBuffer.position(bufferInfo.offset)
+                            outputBuffer.limit(bufferInfo.offset + bufferInfo.size)
 
-                    codec.releaseOutputBuffer(outputBufferIndex, false)
+                            if (trackIndex >= 0) {
+                                muxer.writeSampleData(trackIndex, outputBuffer, bufferInfo)
+                            } else {
+                                appLogger.e("Track index is not set. Skipping writeSampleData.")
+                            }
+                        }
+
+                        codec.releaseOutputBuffer(outputBufferIndex, false)
+                        retryCount = 0
 
-                    if (bufferInfo.flags and MediaCodec.BUFFER_FLAG_END_OF_STREAM != 0) {
-                        sawOutputEOS = true
+                        if (bufferInfo.flags and MediaCodec.BUFFER_FLAG_END_OF_STREAM != 0) {
+                            sawOutputEOS = true
+                        }
+                    }
+                    outputBufferIndex == MediaCodec.INFO_TRY_AGAIN_LATER -> {
+                        retryCount++
+                        Thread.sleep(RETRY_DELAY_IN_MILLIS)
```

Comment:
Question: Is there a reason we can't use `Coroutine.delay()` here instead ? 🤔 
Note: Also the 10 Mills maybe is too little ?